code/modules/mob/living/simple_animal/hostile/zombie.dm

```diff
@@ -0,0 +1,156 @@
+/mob/living/simple_animal/hostile/zombie
+	name = "zombie"
+	desc = "When Observe is full, the dead shall walk the station."
+	icon = 'icons/mob/human.dmi'
+	icon_state = "zombie_s"
+	icon_living = "zombie_s"
+	icon_dead = "zombie_dead"
+	turns_per_move = 5
+	emote_see = list("groans")
+	a_intent = "harm"
+	maxHealth = 120
+	health = 120
+	speed = 1
+	harm_intent_damage = 8
+	melee_damage_lower = 20
+	melee_damage_upper = 20
+	attacktext = "claws"
+	attack_sound = 'sound/hallucinations/growl1.ogg'
+	atmos_requirements = list("min_oxy" = 0, "max_oxy" = 0, "min_tox" = 0, "max_tox" = 0, "min_co2" = 0, "max_co2" = 5, "min_n2" = 0, "max_n2" = 0)
+	minbodytemp = 0
+	maxbodytemp = 350
+	unsuitable_atmos_damage = 10
+	environment_smash = 1
+	robust_searching = 1
+	stat_attack = 2
+	gold_core_spawnable = 1
+	faction = list("zombie")
+	var/mob/living/carbon/human/stored_corpse = null
+	butcher_results = list(/obj/item/weapon/reagent_containers/food/snacks/meat/slab/human/mutant/zombie = 3)
+	see_invisible = SEE_INVISIBLE_MINIMUM
```

Comment:
Oh it's because the darkness overlay has an invisibility of 20 and this define is 5.